projects/concurrency/list/list.go

```diff
@@ -0,0 +1,86 @@
+package list
+
+type Entry[K comparable, V any] struct {
+	key         K
+	Value       V
+	IsRead      bool //Question 2
+	EntriesRead int  //Question 3
+}
+type Node[K comparable, V any] struct {
+	Data Entry[K, V]
```

Comment:
Is "Has this value been read" a general concept of all lists? I'd argue not - it's a caching concept, not a list concept.

Which suggests our `List` type shouldn't know about the `Entry` type - our `List` should just have generic `V`s, and our `Cache` type should make a `List` where the values are `Entry` values (the type of which should be defined in the cache package not the list one).